### PR TITLE
Fix keyword vpd

### PR DESCRIPTION
**Pull request by Alpana07**
Test: Tested code with old keyword vpd.

VPD collected in vector
Start Tag is - 82
DBG:description size : 10
Start Tag found is - 84
DBG:parser Entered
FN :  01KU724
PN : 01KU724
SN : YH30BG78B014
CC : 2D37
CE : 1
VZ : 03
MF :
SM : ?P2-D4      2SSC?P2-D5      2SSC?P2-D7      2SSC?P2-D8      2SSC
FL : P2
B2 : P`sr
WI :
DOne with keywords...
DBG: Accumulated value - 50
DBG:calculated checksum- b0
DBG: Checksum from vpd -b0

Change-Id: Icb4ada91158ea5a658f81246e7eaa3394227c7cb

### CODE REVIEW

File: src/keyword_vpd_parser.cpp [hunk 1]
```diff
@@ -48,11 +50,14 @@ types::VPDMapVariant KeywordVpdParser::parse()
             throw(DataException("Invalid Small resource type End"));
         }
 
+        validateChecksum(checkSumStart, checkSumEnd);
```
Comment (branupama) [inline src/keyword_vpd_parser.cpp:53]: move the incrementing two bytes out of validateChecksum, so it is visible incrementing two bytes to check the  VPD end Tag.

Comment (SunnySrivastava1984) [reply src/keyword_vpd_parser.cpp:53]: I agree, validate should only validate. Also could you please update the parameters as per the guidelines?

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: src/keyword_vpd_parser.cpp (hunk 1 above)
Comment (Alpana07) [reply src/keyword_vpd_parser.cpp:53]: okay

**Comment by branupama** [comment]
Empty input Vector test case is failing for Keyword VPD parser.

**Comment by Alpana07** [comment]
Empty input Vector has been handled.

File: src/keyword_vpd_parser.cpp [hunk 2]
```diff
@@ -15,6 +15,11 @@ types::VPDMapVariant KeywordVpdParser::parse()
 {
     try
     {
+        if (m_keywordVpdVector.empty())
+        {
+            throw(                
+                DataException("VPD Vector is empty"));
```
Comment (branupama) [inline src/keyword_vpd_parser.cpp:21]: Should we have "Keyword" on the log ?
Keyword VPD vector is empty

**Comment by Alpana07** [comment]
Jenkins run tests please

**Comment by Alpana07** [comment]
Jenkins run tests please

**Comment by Alpana07** [comment]
Jenkins run tests please

File: include/keyword_vpd_parser.hpp [hunk 3]
```diff
@@ -56,12 +56,13 @@ class KeywordVpdParser : public ParserInterface
      * Finding the 2's complement of sum of all the
      * keywords,values and large resource identifier string.
      *
-     * @param checkSumStart - vpd iterator pointing at checksum start value
-     * @param checkSumEnd - vpd iterator pointing at checksum end value
+     * @param[in] i_checkSumStart - vpd iterator pointing at checksum start
```
Comment (SunnySrivastava1984) [inline include/keyword_vpd_parser.hpp:59]: abbreviations should be in all caps.
VPD at some place is written as vpd at some place it is VPD.
Please have same across file "VPD".

File: src/keyword_vpd_parser.cpp [hunk 4]
```diff
@@ -48,12 +53,19 @@ types::VPDMapVariant KeywordVpdParser::parse()
...
         // Check vpd end Tag.
         if (*m_vpdIterator != constants::KW_VPD_END_TAG)
         {
             throw(DataException("Invalid Small resource type."));
         }
-        return kwValMap;
+
+        return l_kwValMap;
     }
     catch (const DataException& ex)
```
Comment (SunnySrivastava1984) [inline src/keyword_vpd_parser.cpp:70]: Though unrelated to this commit but as we are refactoring, I would like you to take this change in this PR.
This catch here does not make any sense as it is just logging and re-throwing the same exception.
Either:
a) Handle exception here itself or,
b) Handle it at the calling place only.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: src/keyword_vpd_parser.cpp (hunk 4 above)
Comment (Alpana07) [reply src/keyword_vpd_parser.cpp]: Handling it at the calling place.

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**